File: Search_Algo/binarySearch.cpp
#include<bits/stdc++.h>
using namespace std;
int binarySearch(int arr[],int l,int r,int x)
{
    if(r >= l)
    {
        int mid = l + (r-1)/2;
        if (arr[mid] == x)
            return mid;
        if(arr[mid] > x)
            return binarySearch(arr,l,mid-1,x);    //left
        return binarySearch(arr,mid+1,r,x); //right
    }
    return -1;
}
int main()
{
    int n;
    cout << "Enter the size of array: \n";
    cin >> n;
    int arr[n];
    cout << "Enter the Elements of array:\n";
    for(int i=0;i<n;i++)
    {
        cin >> arr[i];
    }
    int x;
    cout << "Enter the element you are searching for :\n";
    cin >> x;
    int result = binarySearch(arr,0,n-1,x);
    (result == -1) ? cout << "Element is not present in array"
                    :cout << "Element is present at index " << result;
}

File: Search_Algo/binarySearch.py
# Binary Search
def binary_search(arr,l,r,x):
    while l<=r:
        mid = l + (r-1)//2

        if arr[mid] == x:
            return mid
        elif arr[mid] > x : #left side
            l = mid + 1
        else:   #right side
            r = mid - 1
    
    return -1




size = int(input('Enter the size of array : '))
arr = []
for i in range (0,size):
    ele = int(input("Enter the elements of array : \n"))
    arr.append(ele)
n = len(arr)
print(arr)
x = int(input('Enter the Element you are searching for:\n'))
result = binary_search(arr,0,len(arr) -1 ,x)
if result == -1:
    print("Element not found!!")
else:
    print("Element was found  \n" + "at index : ",result)
